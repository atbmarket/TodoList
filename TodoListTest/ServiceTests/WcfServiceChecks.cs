using ApprovalTests;
using ApprovalTests.Reporters;
using SoapService;
using System;
using System.Linq;
using System.Net.Http;
using System.ServiceModel;
using System.ServiceModel.Description;
using System.Text;
using System.Text.RegularExpressions;
using Xunit;
#pragma warning disable IDE1006 // Naming Styles

[assembly: FrontLoadedReporter(typeof(BeyondCompare4Reporter))]
namespace ServiceTests
{
    public class WcfServiceChecks
    {
        [Fact]
        [UseReporter(typeof(BeyondCompare4Reporter))]
        public void ensure_wsdl_is_stable()
        {
            const string address = "http://localhost:11111/hello";
            using (var host = CreateAndOpenHost(address))
            {
                var wsdl = new HttpClient().GetStringAsync($"{address}?singleWsdl").Result;

                Approvals.VerifyXml(wsdl);
            }
        }

        private static ServiceHost CreateAndOpenHost(string address)
        {
            var baseAddress = new Uri(address);
            var host = new ServiceHost(typeof(MyService), baseAddress);
            AddMetadata(host);
            host.Open();
            return host;
        }

        private static void AddMetadata(ServiceHost host)
        {
            var smb = new ServiceMetadataBehavior
            {
                HttpGetEnabled = true
            };
            smb.MetadataExporter.PolicyVersion = PolicyVersion.Policy15;
            host.Description.Behaviors.Add(smb);
        }
    }
}
#pragma warning restore IDE1006 // Naming Styles
