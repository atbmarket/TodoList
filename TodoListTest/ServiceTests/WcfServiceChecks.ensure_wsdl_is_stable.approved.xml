<wsdl:definitions name="MyService" targetNamespace="http://tempuri.org/" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:wsap="http://schemas.xmlsoap.org/ws/2004/08/addressing/policy" xmlns:wsa10="http://www.w3.org/2005/08/addressing" xmlns:tns="http://tempuri.org/" xmlns:msc="http://schemas.microsoft.com/ws/2005/12/wsdl/contract" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/" xmlns:wsx="http://schemas.xmlsoap.org/ws/2004/09/mex" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:wsam="http://www.w3.org/2007/05/addressing/metadata" xmlns:wsa="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:wsp="http://www.w3.org/ns/ws-policy" xmlns:wsaw="http://www.w3.org/2006/05/addressing/wsdl" xmlns:soap12="http://schemas.xmlsoap.org/wsdl/soap12/" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <wsdl:types>
    <xs:schema elementFormDefault="qualified" targetNamespace="http://tempuri.org/" xmlns:xs="http://www.w3.org/2001/XMLSchema">
      <xs:element name="DoWork">
        <xs:complexType>
          <xs:sequence />
        </xs:complexType>
      </xs:element>
      <xs:element name="DoWorkResponse">
        <xs:complexType>
          <xs:sequence />
        </xs:complexType>
      </xs:element>
    </xs:schema>
  </wsdl:types>
  <wsdl:message name="IMyService_DoWork_InputMessage">
    <wsdl:part name="parameters" element="tns:DoWork" />
  </wsdl:message>
  <wsdl:message name="IMyService_DoWork_OutputMessage">
    <wsdl:part name="parameters" element="tns:DoWorkResponse" />
  </wsdl:message>
  <wsdl:portType name="IMyService">
    <wsdl:operation name="DoWork">
      <wsdl:input wsam:Action="http://tempuri.org/IMyService/DoWork" message="tns:IMyService_DoWork_InputMessage" />
      <wsdl:output wsam:Action="http://tempuri.org/IMyService/DoWorkResponse" message="tns:IMyService_DoWork_OutputMessage" />
    </wsdl:operation>
  </wsdl:portType>
  <wsdl:binding name="BasicHttpBinding_IMyService" type="tns:IMyService">
    <soap:binding transport="http://schemas.xmlsoap.org/soap/http" />
    <wsdl:operation name="DoWork">
      <soap:operation soapAction="http://tempuri.org/IMyService/DoWork" style="document" />
      <wsdl:input>
        <soap:body use="literal" />
      </wsdl:input>
      <wsdl:output>
        <soap:body use="literal" />
      </wsdl:output>
    </wsdl:operation>
  </wsdl:binding>
  <wsdl:service name="MyService">
    <wsdl:port name="BasicHttpBinding_IMyService" binding="tns:BasicHttpBinding_IMyService">
      <soap:address location="http://localhost:11111/hello" />
    </wsdl:port>
  </wsdl:service>
</wsdl:definitions>